### ISSUE TITLE
Cache description: syntax highlighting for HTML Source view

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#16300]
Ticket #719845
cgeo version 2024.10.27

In c:geo's description style option "HTML Source", the HTML is displayed unformatted and without any color coding.

A user on support wishes to at least add color coding for HTML comments (e.g. displaying HTML comments like `<!-- this is a comment -->` in a different color eg blue).  

A more throughout immpllementation would also add some formatting (means: inserting line feeds and intends so that HTML is better readable) and color coding (e.g. also mark tag names, attribute names and such in different colors). Much like modern browsers do it in their inspection views - maybe a tad less complex :-)

![image](https://github.com/user-attachments/assets/e9785817-2e6a-4dc0-8380-fa8172526966)

![image](https://github.com/user-attachments/assets/34358504-2d03-48e8-a03f-3e7a3e7b1259)

**Comment by moving-bits** [comment]
The following links describe an interesting approach: Instead of implementing our own highlighting or integrating a separate Java library for it, use a well-established JS syntax highlighting library in conjunction with built-in WebView component:

- https://hossainkhan.medium.com/source-code-syntax-highlighting-on-android-taking-full-control-b704fd4bd8ee
- https://github.com/hossain-khan/android-syntax-highlighter

**Comment by eddiemuc** [comment]
> The following links describe an interesting approach: Instead of implementing our own highlighting or integrating a separate Java library for it, use a well-established JS syntax highlighting library in conjunction with built-in WebView component:
> 
> * https://hossainkhan.medium.com/source-code-syntax-highlighting-on-android-taking-full-control-b704fd4bd8ee
> * https://github.com/hossain-khan/android-syntax-highlighter

Usage of "WebView" is the no-no-trigger for me personally here. I don't have good experience with this one  :-)

**Comment by moving-bits** [comment]
> Usage of "WebView" is the no-no-trigger for me personally here. I don't have good experience with this one :-)

:smile: 

Ok, just was an idea for not having to integrate yet another library if we want to pursue syntax highlighting. But there may be lightweight alternatives (hopefully).

**Comment by eddiemuc** [comment]
Implemented with PR #16315 

While looking for libs to help with this I realized that c:geo already incorporates [JSoup](https://jsoup.org/) for other purposes, so it was the logical choice not to use yet another second lib. Turns out JSoup is great in doing the HTML cleaning and the pretty printing, but lacks support to add custom colors easily to pretty-printed HTML. So that part had to be done manually.
